### DISCUSSION TITLE
Tag textobjects

### DISCUSSION THREAD

**Discussion by christianholman** [71/dance#315] [Ideas]
Was wondering if there's any work done on this already? I'm using a helix-like keybinding settings, and being able to match in/around tags for JSX etc would be amazing. I've tried to emulate the effect with commands to no luck. Seriously the only thing that the vim extensions have over the other modal editors for me right now. I implemented it in meow-edit for emacs by using Treesitter jsx nodes so that could be an approach I imagine.

Here's the tagMatcher.ts from VsCodeVim:
https://github.com/VSCodeVim/Vim/blob/master/src/common/matching/tagMatcher.ts

**Comment by 71** [comment]
If you're fine with simple RegExp-based selections, you can use the `dance.seek.object` command with e.g. `<[a-z-]+.*>(?#inner)</[a-z-]+>` (you can test this by pressing `Alt-A` or `Alt-I`, selecting a custom input with `c`, and entering the pattern).

For more advanced usages, Dance can interface with Tree Sitter but you need to build and install the [Tree Sitter API](https://github.com/71/vscode-tree-sitter-api) extension first. Also, the only commands which use Tree Sitter are `dance.seek.syntax.*.experimental`. Ideally Dance would support [Helix queries](https://github.com/helix-editor/helix/tree/6ab774da0b473c36a437d90abcdac3558353d32e/runtime/queries), but this isn't the case yet (and it looks like Helix does not have textobjects for HTML yet anyway).

**Reply by christianholman** [reply]
Awesome, the regex method is a good start, I'lll have a look at integrating the tree sitter api ! Thank you so much.
